staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/helpers_test.go

```diff
@@ -249,3 +249,204 @@ func TestSetMetaDataLabel(t *testing.T) {
 		}
 	}
 }
+
+func TestFieldsV1MarshalJSON(t *testing.T) {
+	for _, tc := range []struct {
+		FieldsV1 FieldsV1
+		Want     []byte
+		Error    string
+	}{
+		{
+			FieldsV1: FieldsV1{},
+			Want:     []byte(`null`),
+		},
+		{
+			FieldsV1: FieldsV1{Raw: []byte{}},
+			Want:     []byte{},
+		},
+		{
+			FieldsV1: FieldsV1{Raw: []byte{0xf6}}, // null
+			Want:     []byte(`null`),
+		},
+		{
+			FieldsV1: FieldsV1{Raw: []byte{0x30}},
+			Want:     []byte{0x30}, // Valid CBOR encoding of -17 and JSON encoding of 0!
```

Comment:
this doesn't look like JSON, shouldn't we get json out?

Reply:
It is valid CBOR negative integer -17 and also valid JSON Number 0, but it's neither a valid CBOR FieldsV1 nor a valid JSON FieldsV1 because it is not a JSON Object/null or a CBOR map/tag/null.

It returns the JSON Number 0 which is the existing behavior for this case.